mne/preprocessing/tests/test_maxwell.py

```diff
@@ -128,11 +140,21 @@ def test_maxwell_filter_additional():
     raw_sss.save(test_outname)
     raw_sss_loaded = Raw(test_outname, preload=True, proj=False,
                          allow_maxshield=True)
+
     # Some numerical imprecision since save uses 'single' fmt
     assert_allclose(raw_sss_loaded._data[:, :], raw_sss._data[:, :],
                     rtol=1e-6, atol=1e-20)
 
-    # Test covariance calculation XXX add this
+    # Test rank of covariance matrices for raw and SSS processed data
+    cov_raw = compute_raw_data_covariance(raw)
+    cov_sss = compute_raw_data_covariance(raw_sss)
+
+    scalings = None
+    cov_raw_rank = _estimate_rank_meeg_cov(cov_raw['data'], raw.info, scalings)
+    cov_sss_rank = _estimate_rank_meeg_cov(cov_sss['data'], raw_sss.info,
+                                           scalings)
 
+    assert_equal(cov_raw_rank, raw.info['nchan'])
+    assert_equal(cov_sss_rank, maxwell.get_num_moments(int_order, 0))
```

Comment:
@dengemann, @Eric89GXL, does this look correct? For raw and SSS processed data, I'm checking that the covariance rank equals the number of channels and the order of the internal multipolar moment space, respectively.